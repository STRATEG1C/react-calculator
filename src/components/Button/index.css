.button {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 80px;
    height: 80px;
    margin: 10px;
    font-size: 18px;
    border-radius: 50%;
    background: beige;
    border: 1px solid #ebebeb;
    cursor: pointer;
    transition: background-color ease .2s;
}

.button:last-child {
    margin-right: 0;
}

.button.invisible {
    visibility: hidden;
}

.button:hover {
    background: #d0d0b4;
}

.button.operator {
    background: #ebebeb;
}

.button.operator:hover {
    background: #d5d5d5;
}

.button.operator-equal {
    background: #c61a45;
    color: #fff;
}
