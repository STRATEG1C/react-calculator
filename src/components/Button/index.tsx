import React from 'react';
import './index.css';

interface Props {
  text: string;
  value: string;
  onClick: (value: string) => void;
  className?: string;
}

export const Button: React.FC<Props> = (props: Props) => {
  const { text, value, onClick, className = '' } = props;

  const onClickHandler = () => onClick(value);

  return <div className={`button ${className}`} onClick={onClickHandler}>{text}</div>
}

export default Button;
