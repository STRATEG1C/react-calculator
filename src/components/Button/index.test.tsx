import { render } from '@testing-library/react';
import Button from './index';
import userEvent from '@testing-library/user-event';

describe('Test Button component', () => {
  const BUTTON_TEXT = 'abc';
  const BUTTON_VALUE = 'abc';

  const onClickStub = jest.fn();

  it('Should be rendered', () => {
    const { getByText } = render(
      <Button
        text={BUTTON_TEXT}
        value={BUTTON_VALUE}
        onClick={() => false}
      />
    );

    const button = getByText(BUTTON_TEXT);

    expect(button).toBeInTheDocument();
    expect(button.textContent).toBe(BUTTON_TEXT);
    expect(button).toMatchSnapshot();
  });

  it('Should react on click', () => {
    const { getByText } = render(
      <Button
        text={BUTTON_TEXT}
        value={BUTTON_VALUE}
        onClick={onClickStub}
      />
    );

    const button = getByText(BUTTON_TEXT);

    userEvent.click(button);

    expect(onClickStub).toHaveBeenCalledWith(BUTTON_VALUE);
  });
});
