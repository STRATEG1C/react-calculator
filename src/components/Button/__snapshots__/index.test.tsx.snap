// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component Should be rendered 1`] = `
<div
  class="button "
>
  abc
</div>
`;
